### ISSUE TITLE
incorrect values in component's state w/ "no-local-state" config

### ISSUE THREAD

**Issue by jack-james** [omcljs/om#335]
minimal case:

```clojure
(ns nolocal-test.core
  (:require [om.core :as om]
            [om.dom :as dom]))

(enable-console-print!)

(def app-state (atom {:value "Hello World!"}))

(defn input [{:keys [value]} _ _]
  (om/component (dom/input #js {:defaultValue value :type "text"})))

(defn editable [{:keys [value] :as data} owner _]
  (reify
    om/IInitState (init-state [_] {:editing? false})
    om/IRenderState

    ;; with "no-local-state" config, clicking the span results in two renders
    ;; after the 2nd render, editing? is nil, so the input disappears immediately
    ;; without "no-local-state" config, behavior is as expected
    ;; (input is rendered when the span is clicked)

    (render-state [_ {:keys [editing?]}]
      (if editing?
        (om/build input data)
        (dom/span #js {:onClick #(om/set-state! owner :editing? true)} value)))))

(om/root
  (fn [data _ _] (om/component (om/build editable data)))
  app-state
  {:instrument
           (fn [f cursor m]
             (om/build* f cursor
               (assoc m :descriptor
                        (om/no-local-descriptor
                          om/no-local-state-methods))))
   :target (. js/document (getElementById "app"))})
```
clicking the span should result in a value of true rather than nil for editing?

**Comment by swannodette** [comment]
Warning `no-local-state` users should expect some churn. React 0.13 now hides `_rootNodeId`.

**Comment by swannodette** [comment]
Will revisit this one.

**Comment by swannodette** [comment]
Because of our reliance on React there's no good way to accomplish good externalized local state anymore. Om Next will have a better solution that avoids all the problems we have encountered due to React internal changes.

**Comment by petterik** [comment]
I just got something working (with issues) for react v0.13: 9179a8aede83c22de34335b0fda58f7314e0a9bc

Fix: Scope issue in componentDidMount. Make the state var which is defined in getInitialState available to componentDidMount.

Issue: There was still an issue with local state components not being reloaded when I swapped the app-state with only local state differences. Probably because no component is watching the :state-map path in the cursor?

@swannodette: Let me know if you want to fix the scoping issue in the experimental componentDidMount, and I'll send you a pull request similar to 9179a8aede83c22de34335b0fda58f7314e0a9bc but without all comments.
